pkg/template/resource.go

```diff
@@ -52,11 +52,22 @@ func ResolveTrigger(trigger triggersv1.EventListenerTrigger, getTB getTriggerBin
 	if err != nil {
 		return ResolvedTrigger{}, fmt.Errorf("failed to resolve bindings: %w", err)
 	}
-	tt, err := getTT(context.Background(), trigger.Template.Name, metav1.GetOptions{})
-	if err != nil {
-		return ResolvedTrigger{}, fmt.Errorf("error getting TriggerTemplate %s: %w", trigger.Template.Name, err)
+
+	var resolvedTT *triggersv1.TriggerTemplate
+	if trigger.Template.Spec != nil {
+		resolvedTT = &triggersv1.TriggerTemplate{
+			ObjectMeta: metav1.ObjectMeta{}, // Unsed. TODO: Just return Specs from here.
+			Spec:       *trigger.Template.Spec,
+		}
+	} else {
+		ttName := trigger.Template.Name
+		resolvedTT, err = getTT(context.Background(), ttName, metav1.GetOptions{})
```

Comment:
We should plumb the context through, otherwise we won't respect timeouts/cancels applied to the context.

(I realize this is carry over from before - your choice if you want to handle it in this PR or if you want to differ to a different PR).

Reply:
yeah, this was recently added due to an upgrade to k8s v1.18 which changed the interfaces for clients to accept a context. 
I think I'd want to do this in a different PR (probably using an lister cache -- so that we are not calling the API server for each event).